src/Toolchain/ToolchainProvider.ts

```diff
@@ -316,27 +316,29 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
   }
 
   public _run(cfg: string): boolean {
+    const [activeToolchainEnv, activeToolchain] =
+      this.checkAvailableToolchain();
+    if (activeToolchainEnv === undefined || activeToolchain === undefined) {
```

Comment:
```suggestion
    if (!activeToolchainEnv || !activeToolchain) {
```

FYI, this is easy and shorter synonym.

Reply:
Thank you for your advice~!! It's better then before 😄 I'll apply it

Reply:
```js
    if (activeToolchainEnv === undefined || activeToolchain === undefined) {
```
and
```js
    if (!activeToolchainEnv || !activeToolchain) {
```
are different as I understand. The former returns `true` only `undefined` while the latter returns `true`, not only for `undefined`, but also for 0, empty string("", ''), NaN, null.

Reply:
@glistening That's right. I meant that here it would mean the same. Usually returning `0, empty string("", ''), NaN, null` from functions mean the similar result, so I take this abbreviation. :-D